### PR TITLE
CB-10327 We should wait for the flow to finish by deafult. The await(…

### PR DESCRIPTION
**Pull request by topolyai5**
…) will wait until the flow finished and after that will check the status. I have alse removed the await(), awaitForFlow(), and status checker redundancy because these are very hard to maintainance.

See detailed description in the commit message.

### CODE REVIEW

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/ResourceAwait.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import com.sequenceiq.it.cloudbreak.MicroserviceClient;
+import com.sequenceiq.it.cloudbreak.context.RunningParameter;
+import com.sequenceiq.it.cloudbreak.context.TestContext;
+import com.sequenceiq.it.cloudbreak.dto.CloudbreakTestDto;
+import com.sequenceiq.it.cloudbreak.log.Log;
+import com.sequenceiq.it.cloudbreak.util.ResponseUtil;
+
+@Component
+public class ResourceAwait {
```
Comment (aszegedi) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/ResourceAwait.java:18]: What is the purpose of this, where it is going to be used?

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/CloudbreakFailedChecker.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.ProcessingException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sequenceiq.it.cloudbreak.exception.TestFailException;
+
+public class CloudbreakFailedChecker<T extends WaitObject> extends ExceptionChecker<T> {
```
[the remark this answers is not in the document]
Comment (topolyai5) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/CloudbreakFailedChecker.java:13]: This name is ok for me

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/CloudbreakOperationChecker.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.Map;
+
+import javax.ws.rs.ProcessingException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sequenceiq.it.cloudbreak.exception.TestFailException;
+
+public class CloudbreakOperationChecker<T extends WaitObject> extends ExceptionChecker<T> {
```
[the remark this answers is not in the document]
Comment (topolyai5) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/CloudbreakOperationChecker.java:12]: This name is ok for me

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/CloudbreakTerminationChecker.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.ProcessingException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sequenceiq.it.cloudbreak.exception.TestFailException;
+
+public class CloudbreakTerminationChecker<T extends WaitObject> extends ExceptionChecker<T> {
```
[the remark this answers is not in the document]
Comment (topolyai5) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/CloudbreakTerminationChecker.java:15]: This name is ok for me

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/ResourceAwait.java (hunk 1 above)
Comment (topolyai5) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/ResourceAwait.java:18]: we call this from the TestContext in the awaitForFlow()

**Review by aszegedi — APPROVED** [review]

**Pull request merged**